src/Database/QueryCompiler.php

```diff
@@ -211,9 +212,15 @@ protected function _buildJoinPart($parts, $query, $generator)
     {
         $joins = '';
         foreach ($parts as $join) {
+            $subquery = $join['table'] instanceof Query || $join['table'] instanceof QueryExpression;
             if ($join['table'] instanceof ExpressionInterface) {
-                $join['table'] = '(' . $join['table']->sql($generator) . ')';
+                $join['table'] = $join['table']->sql($generator);
```

Comment:
Interesting one, I haven't tried that specifically. One of the tests are failing so if that would case a failure in such a case it would require some more tests for that.